### PR TITLE
Add Icon Dropdown Component

### PR DESCRIPTION
**Pull request by dav-is**
The `IconDropdown` component is the same as a `Dropdown`, except the it houses font awesome icons for selection. This adds a cleaner way for users to select a font awesome icon. I noticed in groups
# Usage:

`selection` the icon that is currently selected. Default value of `question`
`icons` the set of icons to display for selection. Possible values: "all", "social", or an array of icon names. Default value of `all`.

```
IconDropdown.component({selection: this.selection, icons: 'social'})
```
# Screenshot:

screenshot

### CODE REVIEW

**Comment by Luceos** [comment]
Out of curiosity what is the effect on loading times with and without the dropdown? Did you check that?

Nice addition.

**Comment by jordanjay29** [comment]
Love this, let's get it in. I have an extension I'm working on that could soon make use of it, and having it as a Flarum component would be perfect.

**Comment by Luceos** [comment]
@flarum/core wouldn't it make more sense to start classifying the components too (much as in the backend side) so abstracts should move to a separate directory and extended components to another? This might help in the future.

**Comment by franzliedke** [comment]
Don't we have a lib folder for that sort of thing?

**Comment by Luceos** [comment]
Well the lib is where all shared code is placed, so those used by forum and admin. If I'm not mistaken @tobscure ?

**Comment by tobscure** [comment]
Nice! Will get this in with some modification when I have time.

I don't see the need for component sub-folders, I think it would be somewhat arbitrary.

File: js/lib/components/IconDropdown.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        'social':
+            ["globe", 'amazon', 'angellist', 'apple', 'behance', 'bitbucket', 'codepen', 'connectdevelop', 'dashcube', 'delicious', 'deviantart', 'digg', 'dribbble', 'dropbox', 'drupal', 'facebook', 'flickr', 'foursquare', 'get-pocket', 'git', 'github', 'github-alt', 'gittip', 'google', 'google-plus', 'google-wallet', 'gratipay', 'hacker-news', 'instagram', 'ioxhost', 'joomla', 'jsfiddle', 'lastfm', 'leanpub', 'linkedin', 'meanpath', 'medium', 'odnoklassniki', 'opencart', 'pagelines', 'paypal', 'pied-piper-alt', 'pinterest-p', 'qq', 'reddit', 'renren', 'sellsy', 'share-alt', 'shirtsinbulk', 'simplybuilt', 'skyatlas', 'skype', 'slack', 'slideshare', 'soundcloud', 'spotify', 'stack-exchange', 'stack-overflow', 'steam', 'stumbleupon', 'tencent-weibo', 'trello', 'tripadvisor', 'tumblr', 'twitch', 'twitter', 'viacoin', 'vimeo', 'vine', 'vk', 'wechat', 'weibo', 'weixin', 'whatsapp', 'wordpress', 'xing', 'y-combinator', 'yelp', 'youtube-play' ],
```
Comment (franzliedke) [inline js/lib/components/IconDropdown.js]: Please extract both lists to variables (put them at the beginning of the file, they're only scoped to this module anyway if you don't export them).
I'd prefer having one item per line - even if that gets very long, it's easier to see what's going on and diffs will be much easier to read.

Also, you don't need to wrap the keys in strings. :)

File: less/lib/Dropdown.less [hunk 2]
```diff
@@ -79,6 +79,49 @@
...
+    -webkit-background-clip: padding-box;
+    background-clip: padding-box;
+    border: 1px solid rgba(0,0,0,.15);
+    -webkit-transform: none !important;
+    transform: none !important;
+    -webkit-transition: none !important;
+    -moz-transition: none !important;
+    -o-transition: none !important;
+    transition: none !important;
+    > li > div {
```
Comment (franzliedke) [inline less/lib/Dropdown.less]: Are these very specific selectors necessary?

Also, what's our policy on browser prefixes, @tobscure ?

Comment (dav-is) [inline less/lib/Dropdown.less]: They are necessary. They overwrite the original Dropdown CSS.

**Comment by dav-is** [comment]
I'm vacationing in China currently and don't have time to work on this. It would be great if you all could fix it up.

File: less/lib/Dropdown.less (hunk 2 above)
Comment (franzliedke) [inline less/lib/Dropdown.less]: That's ... too bad. :(

Comment (jordanjay29) [inline less/lib/Dropdown.less]: Why is it necessary to reuse the same class?

File: less/lib/Dropdown.less [hunk 3]
```diff
@@ -79,6 +79,49 @@
...
 }
+.Dropdown-menu--below {
+    padding-left: 10px !important;
+    padding-top: 10px !important;
+    bottom: inherit !important;
+    position: absolute !important;
+    top: 100% !important;
+    overflow: auto;
+    max-height: 45vh !important;
+    min-width: 189px !important;
```
Comment (franzliedke) [inline less/lib/Dropdown.less]: I must say that all these `!important` are a bad sign, too. It means we have too specific selectors somewhere else.

Comment (tobscure) [inline less/lib/Dropdown.less]: Yep, one of the things I will modify.

File: less/lib/Dropdown.less (hunk 2 above)
Comment (tobscure) [inline less/lib/Dropdown.less]: @franzliedke we have mixins which automatically apply browser prefixes. See less/lib/mixins/vendor-prefixes.less

Comment (dav-is) [inline less/lib/Dropdown.less]: @jordanjay29 the dropdown component adds the .Dropdown-menu class no matter what

**Comment by ardacebi** [comment]
@dav-is 

- How many font awesome icons does this picker has? 
- Does it have all or just the common ones? 
- How should users input a font awesome icon that it isn’t on the picker if it only has the common icons?
- Does this use FA 4 or FA 5?

Many thanks.

**Pull request closed without merging**